src/main/java/uk/gov/hmcts/reform/civil/service/citizenui/DashboardClaimInfoService.java

```diff
@@ -39,16 +39,30 @@ public List<DashboardClaimInfo> getClaimsForClaimant(String authorisation, Strin
     public List<DashboardClaimInfo> getClaimsForDefendant(String authorisation, String defendantId) {
         List<DashboardClaimInfo> ocmcClaims = claimStoreService.getClaimsForDefendant(authorisation, defendantId);
         List<DashboardClaimInfo> ccdCases = getCases(authorisation);
-        return Stream.concat(ocmcClaims.stream(), ccdCases.stream()).collect(Collectors.toList());
+
+        return Stream.concat(ocmcClaims.stream(), ccdCases.stream())
+            .sorted(this::compareClaimByCreateDate)
+            .collect(Collectors.toList());
     }
 
-    private List<DashboardClaimInfo> getCases(String authorisation) {
-        Query query = new Query(QueryBuilders.matchAllQuery(), emptyList(), 0);
-        SearchResult claims = coreCaseDataService.searchCases(query, authorisation);
-        if (claims.getTotal() == 0) {
-            return Collections.emptyList();
+    private int compareClaimByCreateDate(DashboardClaimInfo claim1, DashboardClaimInfo claim2) {
+        if (claim1.getCreatedDate() == null || claim2.getCreatedDate() == null) {
+            return 0;
         }
-        return translateSearchResultToDashboardItems(claims);
+        return claim2.getCreatedDate().compareTo(claim1.getCreatedDate());
+    }
+
+    private List<DashboardClaimInfo> getCases(String authorisation) {
+        List<DashboardClaimInfo> dashboardClaimItems = new ArrayList<>();
+        int totalCases = 0;
+        SearchResult claims;
+        do {
+            Query query = new Query(QueryBuilders.matchAllQuery(), emptyList(), totalCases);
+            claims = coreCaseDataService.searchCases(query, authorisation);
+            dashboardClaimItems.addAll(translateSearchResultToDashboardItems(claims));
+            totalCases += claims.getCases().size();
+        } while (totalCases < claims.getTotal());
```

Comment:
What happens here if there are 0 claims returned by the first query? Also what's the reason for paging the query here rather than just getting all matching claims?

Reply:
Discussed the second point above with Kamil, the paging is in the core search query which we could overload to not specify a page size, but agreed this is out of scope for this story